src/renderer/components/archive/archive-search.js

```diff
@@ -0,0 +1,279 @@
+  top: 0;
+  width: 100%;
+  height: 100%;
+  z-index: 1;
+`;
+
+const Search = styled(Flex)`
+  position: absolute;
+  z-index: 4;
+  top: 50%;
+  left: 50%;
+  width: 50vw;
+  max-width: 400px;
+  transition: transform 0.3s;
+  transform: translate(-50%, -50%);
+  width: 100%;
+`;
+
+const Input = styled(BaseInput)`
+  padding: 20px 10px;
+  margin: ${props =>
+    props.entries.length > 0 || props.searchTerm !== '' ? '0 0 20px 0' : 0};
+`;
+
+/**
+	* Entry-List
+	*/
+const EntryList = styled(Box)`
+  background-color: #fff;
+  padding: var(--spacing-two);
+  border: 0;
+  margin: 0;
+  width: 100%;
+  border-radius: 6px;
+  position: relative;
+  &:after {
+    bottom: 100%;
+    left: 30px;
+    border: solid transparent;
+    content: ' ';
+    position: absolute;
+    pointer-events: none;
+    border-color: rgba(136, 183, 213, 0);
+    border-bottom-color: #fff;
+    border-width: 10px;
+    margin-left: -10px;
+  }
+`;
+
+const NothingFound = styled(EntryList)`
+  color: #999;
+  text-align: center;
+  font-size: 14px;
+`;
+
+const ListItem = styled.div`
+  margin: 0;
+  font-size: 14px;
+  padding: 15px;
+  cursor: pointer;
+  border-bottom: 1px solid #eee;
+
+  &:last-child {
+    border: 0;
+  }
+
+  &:hover {
+    background-color: #00b7ac;
+    color: #fff;
+    p {
+      color: #fff;
+    }
+  }
+`;
+
+const EntryData = styled.div`
+  display: inline-block;
+  margin: -5px 0 0 0;
+`;
+
+const EntryFolder = styled('p')`
+  font-size: 12px;
+  color: #999;
+  margin: 0;
+`;
+
+const Icon = styled.div`
+  float: left;
+  margin: 0 10px 0 0;
+`;
+
+const FolderIcon = styled('img')`
+  width: 22px;
+`;
+
+class ArchiveSearch extends PureComponent {
+  static propTypes = {
+    onSelectEntry: PropTypes.func,
+    getArchive: PropTypes.func,
+    currentArchive: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+    onGroupSelect: PropTypes.func,
+    t: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      visible: false,
+      archive: null,
+      entries: [],
+      groups: [],
+      searchTerm: ''
+    };
+    this.changeInput = this.changeInput.bind(this);
+    this.closeSearch = this.closeSearch.bind(this);
+    this.searchListener = this.searchListener.bind(this);
+    this.highlightSearchResult = this.highlightSearchResult.bind(this);
+  }
+
+  searchListener() {
+    const { getArchive, currentArchive } = this.props;
+
+    if (currentArchive) {
+      const archive = getArchive(currentArchive.id);
+
+      this.setState(state => ({
+        visible: !state.visible,
+        archive
+      }));
+
+      this._input.focus();
+      this._input.select();
+    }
+  }
+
+  closeSearch() {
+    this.setState({
+      visible: false
+    });
+  }
+
+  changeInput(e) {
+    const value = e.target.value;
+    const entries = value
+      ? this.state.archive.findEntriesByProperty('title', value)
+      : [];
+    const groups = value ? this.state.archive.findGroupsByTitle(value) : [];
```

Comment:
Do we need to search groups? I can't really see there being that many groups that they'd need to be included in search results. What's your opinion @sallar @ph1p ?

Reply:
@perry-mitchell I asked a colleague of mine for his opinion on this PR. He said it would be nice to see folders. So I added them. For me only entries are important.

Reply:
Perhaps @sallar has a point on this? :smile: